roles/tests-integration/tests/pool_integration.rs

```diff
@@ -15,7 +15,7 @@ async fn success_pool_template_provider_connection() {
     let tp_addr = common::get_available_address();
     let pool_addr = common::get_available_address();
     let _tp = common::start_template_provider(tp_addr.port()).await;
-    let sniffer = common::start_sniffer(sniffer_addr, tp_addr).await;
+    let sniffer = common::start_sniffer(sniffer_addr, tp_addr, None).await;
```

Comment:
I think we should add a new `intercept_pool_template_provider_connection` to make sure the new `Sniffer` feature behaves as expected

potentially with 2 `Sniffer`s:
- one intercepts `SetupConnection.Success` and replaces with `SetupConnection.Error`
- one asserts `SetupConnection.Error`